### PR TITLE
Message queue consumption prohibition plugin: UT for kafka and related configuration.

### PR DESCRIPTION
**Pull request by lilai23**
【Fix issue】#1379

[Modification content] Add kakfa to prohibit consumption of interceptors, tool classes, controllers and configuration management related UT

[Use case description] Add code UT

[Self-test situation] 1. Local static check passed; 2. Except for the bytecode enhanced declarator and its auxiliary classes, UT's class coverage is 100%, method coverage is 67%, and line coverage is 61%

[Scope of influence] None

### CODE REVIEW

File: sermant-plugins/sermant-mq-consume-prohibition/kafka-1.x-2.x-plugin/src/test/java/com/huaweicloud/sermant/kafka/utils/MarkUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        MarkUtils.setMark(false);
+        Assert.assertFalse(MarkUtils.getMark());
+    }
+
+    /**
+     * 测试setMark
+     */
+    @Test
+    public void testSetMark() {
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-consume-prohibition/kafka-1.x-2.x-plugin/src/test/java/com/huaweicloud/sermant/kafka/utils/MarkUtilsTest.java]: 这俩是不是留一个就行，testSetAndGetMark。最后在remove下吧

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/cache/KafkaConsumerCacheTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * 测试addKafkaConsumer方法
+     */
+    @Test
+    public void testAddKafkaConsumer() {
+        KafkaConsumer<?, ?> mockConsumer = Mockito.mock(KafkaConsumer.class);
+        int hashCode = mockConsumer.hashCode();
+        KafkaConsumerCache.INSTANCE.addKafkaConsumer(mockConsumer);
+
+        Assert.assertTrue(KafkaConsumerCache.INSTANCE.getCache().containsKey(hashCode));
+        Assert.assertEquals(mockConsumer, KafkaConsumerCache.INSTANCE.getCache().get(hashCode).getKafkaConsumer());
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/cache/KafkaConsumerCacheTest.java:65]: 看起来只需要下面的assert就可以了

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    public void testDisableConsumptionWithNoTopic() {
+        KafkaConsumer<?, ?> mockConsumer = Mockito.mock(KafkaConsumer.class);
+        KafkaConsumerWrapper kafkaConsumerWrapper = new KafkaConsumerWrapper(mockConsumer);
+        HashSet<String> originalTopics = new HashSet<>();
+        kafkaConsumerWrapper.setOriginalTopics(originalTopics);
+        Set<String> prohibitionTopics = new HashSet<>();
+        KafkaConsumerController.disableConsumption(kafkaConsumerWrapper, prohibitionTopics);
+        Mockito.verify(mockConsumer, Mockito.times(0)).subscribe(
+                (Collection<String>) Mockito.any());
+        Mockito.verify(mockConsumer, Mockito.times(0)).assign(Mockito.any());
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:70]: 和上面保持一致，加上强转类型

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        kafkaConsumerWrapper.setOriginalTopics(originalTopics);
+        kafkaConsumerWrapper.setAssign(false);
+        Set<String> prohibitionTopics = new HashSet<>();
+        KafkaConsumerController.disableConsumption(kafkaConsumerWrapper, prohibitionTopics);
+        Mockito.verify(mockConsumer, Mockito.times(1)).subscribe(
+                Collections.singletonList("testTopic-1"));
+    }
+
+    /**
+     * 测试subscribe方式originalTopics减去prohibitionTopics的情况
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:91]: 测试subscribe方式originalTopics和prohibitionTopics存在交集的情况

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        kafkaConsumerWrapper.setAssign(false);
+        Set<String> prohibitionTopics = new HashSet<>();
+        prohibitionTopics.add("testTopic-3");
+        KafkaConsumerController.disableConsumption(kafkaConsumerWrapper, prohibitionTopics);
+        Mockito.verify(mockConsumer, Mockito.times(1)).subscribe(Arrays.asList("testTopic-1", "testTopic-2"));
+    }
+
+    /**
+     * 测试assign方式originalTopics减去prohibitionTopics的情况
+     */
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:129]: 测试assign方式originalTopics和prohibitionTopics存在交集的情况

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/cache/KafkaConsumerCacheTest.java (hunk 2 above)
Comment (lilai23) [reply sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/cache/KafkaConsumerCacheTest.java:65]: 不检查上面的 下面assert会报空指针异常

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java (hunk 3 above)
Comment (lilai23) [reply sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:70]: 上面String必须强转，下面是不需要强转的

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java (hunk 4 above)
Comment (lilai23) [reply sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:91]: 这里就是交集的测试

File: sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java (hunk 5 above)
Comment (lilai23) [reply sermant-plugins/sermant-mq-consume-prohibition/consumer-controller/src/test/java/com/huaweicloud/sermant/kafka/controller/KafkaConsumerControllerTest.java:129]: 这里就是交集的测试

File: sermant-plugins/sermant-mq-consume-prohibition/kafka-1.x-2.x-plugin/src/test/java/com/huaweicloud/sermant/kafka/utils/MarkUtilsTest.java (hunk 1 above)
Comment (lilai23) [reply sermant-plugins/sermant-mq-consume-prohibition/kafka-1.x-2.x-plugin/src/test/java/com/huaweicloud/sermant/kafka/utils/MarkUtilsTest.java]: 已修改

**Review by luanwenfei-venus — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**